patches/expo-secure-store/expo-secure-store+14.2.4+001+additional-config-options.patch

```diff
@@ -0,0 +1,1114 @@
+       guard let key = validate(for: key) else {
+         throw InvalidKeyException()
+       }
+ 
+-      return try set(value: value, with: key, options: options)
++      if options.requireAuthentication && options.authOnEveryAction {
++        try await askForAuth(
++          prompt: options.authenticationPrompt
++          ?? "Authentication required"
++        )
++      }
++
++      let result = try set(value: value, with: key, options: options)
++
++      if !result {
++        return AuthType.none.rawValue
++      }
++
++      return getAuthType().rawValue
+     }
+ 
+-    Function("setValueWithKeySync") {(value: String, key: String, options: SecureStoreOptions) -> Bool in
++    Function("setValueWithKeySync") {(value: String, key: String, options: SecureStoreOptions) -> Int in
+       guard let key = validate(for: key) else {
+         throw InvalidKeyException()
+       }
+ 
+-      return try set(value: value, with: key, options: options)
++      let result = try set(value: value, with: key, options: options)
++
++      if !result {
++        return AuthType.none.rawValue
++      }
++
++      return getAuthType().rawValue
+     }
+ 
+     AsyncFunction("deleteValueWithKeyAsync") { (key: String, options: SecureStoreOptions) in
+-      let noAuthSearchDictionary = query(with: key, options: options, requireAuthentication: false)
+-      let authSearchDictionary = query(with: key, options: options, requireAuthentication: true)
+-      let legacySearchDictionary = query(with: key, options: options)
++      let noAuthSearchDictionary = try query(with: key, options: options, requireAuthentication: false)
++      let authSearchDictionary = try query(with: key, options: options, requireAuthentication: true)
++      let legacySearchDictionary = try query(with: key, options: options)
+ 
+       SecItemDelete(legacySearchDictionary as CFDictionary)
+       SecItemDelete(authSearchDictionary as CFDictionary)
+       SecItemDelete(noAuthSearchDictionary as CFDictionary)
+     }
+ 
+-    Function("canUseBiometricAuthentication") {() -> Bool in
+-      #if os(tvOS)
+-      return false
+-      #else
+-      let context = LAContext()
+-      var error: NSError?
+-      let isBiometricsSupported: Bool = context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &error)
++    Function("canUseBiometricAuthentication") { () -> Bool in
++      return areBiometricsEnabled()
++    }
++  }
++
++  private func askForAuth(prompt: String) async throws {
++    let context = LAContext()
++    context.localizedReason = prompt
++    var error: NSError?
++    let isAuthenticationSupported: Bool = context.canEvaluatePolicy(
++      LAPolicy.deviceOwnerAuthentication,
++      error: &error
++    )
++    if error != nil {
++      throw error ?? RuntimeError("No authentication method available")
++    }
++    if !isAuthenticationSupported {
++      throw RuntimeError("No authentication method available")
++    }
```

Comment:
I think this would be a cleaner solution. In this case we can also skip the error value, because in case there is an error the code will enter the `else` block anyways. Besides, we don't use the error message

```suggestion
guard context.canEvaluatePolicy(.deviceOwnerAuthentication, error: nil) else {
  throw RuntimeError("No authentication method available")
}
```